src/input/mod.rs

```diff
@@ -1,18 +1,70 @@
+#[cfg(feature = "async")]
+use crate::WaitableInputSwitch;
 use crate::{ActiveHigh, ActiveLow, InputSwitch, Switch};
-use embedded_hal::digital::v2::InputPin;
+use embedded_hal::digital::{ErrorType, InputPin};
+#[cfg(feature = "async")]
+use embedded_hal_async::digital::Wait;
 
 impl<T: InputPin> InputSwitch for Switch<T, ActiveHigh> {
-    type Error = <T as InputPin>::Error;
+    type Error = <T as ErrorType>::Error;
 
-    fn is_active(&self) -> Result<bool, Self::Error> {
+    fn is_active(&mut self) -> Result<bool, Self::Error> {
         self.pin.is_high()
```

Comment:
Why mutable? Asking the pin for its state should not mutate it.

Reply:
Thank you for the review. The reason for this change (and other similar changes) is the change in the embedded-hal. In 1.0.0 the trait InputPin defines both is_high/is_low methods as taking &mut self. So I have propagated it to the switch-hal level.

There's is a solution for that, of course: interior mutability, and I can rework my patch using that approach. I just didn't want to overcomplicate it. So tell me your preference: would you like me to do it?

Reply:
Ahhhh and now I wish I hadn’t left the working group prior to that crate going 1.0. Give me a moment to go figure out why they made that decision so I can see if it affects the one we make here. Hold tight please. I appreciate your patience with me.

Reply:
Found [the PR where the hal was changed](https://github.com/rust-embedded/embedded-hal/pull/547). 

Long story short, it should never have happened and a strong personality discarded some very wise advice to not make it mutable. I don’t wish to propagate that mistake if it can be avoided. If you don’t mind taking a stab at preserving our existing `&self` API, I would appreciate it.

Reply:
Let me look into this.